src/Bridge/Doctrine/Orm/AbstractPaginator.php

```diff
@@ -0,0 +1,75 @@
+<?php
+
+/*
+ * This file is part of the API Platform project.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace ApiPlatform\Core\Bridge\Doctrine\Orm;
+
+use ApiPlatform\Core\DataProvider\PartialPaginatorInterface;
+use ApiPlatform\Core\Exception\InvalidArgumentException;
+use Doctrine\ORM\Query;
+use Doctrine\ORM\Tools\Pagination\Paginator as DoctrinePaginator;
+
+abstract class AbstractPaginator implements \IteratorAggregate, PartialPaginatorInterface
+{
+    protected $paginator;
+    protected $iterator;
+    protected $firstResult;
+    protected $maxResults;
+
+    /**
+     * @throws InvalidArgumentException
+     */
+    public function __construct(DoctrinePaginator $paginator)
+    {
+        $query = $paginator->getQuery();
+
+        if (null === ($firstResult = $query->getFirstResult()) || null === $maxResults = $query->getMaxResults()) {
+            throw new InvalidArgumentException(sprintf('"%1$s::setFirstResult()" or/and "%1$s::setMaxResults()" was/were not applied to the query.', Query::class));
+        }
```

Comment:
Because these two methods can potentially return `null`. See the Doctrine PHPDoc:

> Returns NULL if {link setFirstResult} was not applied to this query.
> Returns NULL if {link setMaxResults} was not applied to this query.

We always call `Query::setFirstResult()` and `Query::setMaxResults()` before instantiating our `Paginator` class, it's why we never had the bug.